packages/marshal/src/encodePassable.js

```diff
@@ -328,11 +451,17 @@ export const makeEncodePassable = (encodeOptions = {}) => {
     encodeRemotable = (rem, _) => Fail`remotable unexpected: ${rem}`,
     encodePromise = (prom, _) => Fail`promise unexpected: ${prom}`,
     encodeError = (err, _) => Fail`error unexpected: ${err}`,
+    xxx = false,
```

Comment:
We'll need a name for the smaller encoding and this property opting in to it, and ideally also a name for the current encoding. I don't have a good suggestion at the moment, so :bike::house: away!

Reply:
Some thoughts:

- `oldAndBusted` vs `newHawtness`
- `tired` vs `wired`

Reply:
* `escapism` vs `realism`

Reply:
For what it's worth, I don't think there's a fundamental aspect to the new encoding other than translation of rank ordering into lexicographic ordering (a feature which it shares with the current encoding). And the key detail differentiating the two is that the current encoding applies escaping to array elements (causing blowup for deep structures) while the new one applies escaping to strings and transitively to symbols, which are the only types that can contain characters confusable with array encoding delimiters and as scalars are immune to the blowup issue.

"Compact" is accurate, but keep in mind that we're still using text rather than raw octets.

Reply:
So does this support naming these `legacyOrdered` vs `compactOrdered` ? If not, then I don't think I got the thrust of your point.

Reply:
I was documenting my thoughts in case that sparked something. "legacy" vs. "compact" seems fine, and we can cross the "moreCompact" bridge if and when we come to it. I do like the implicit value judgment that wouldn't be present in an alternative pair like e.g. "recursiveEscaping" vs. "scalarEscaping".

Reply:
So `legacyOrdered` vs `compactOrdered` is fine?

Reply:
I would think just `"legacy"` vs. `"compact"`, e.g. `makeEncodePassable({ format: "compact" })`. What significance of "ordered" am I missing?

Reply:
Neither smallcaps nor its predecessor are order preserving. The main feature of both the formats we're naming here is that they are.

Reply:
Ah, so IIUC you're thinking about something like `makeMarshal(convertValToSlot, convertSlotToVal, { serializeBodyFormat: "compactOrdered" })` and wanting "compactOrdered" to be used everywhere. :+1:

Reply:
yes, exactly.